aries_cloudagent/resolver/did_resolver.py

```diff
@@ -35,27 +29,30 @@ async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
         """Retrieve did doc from public registry."""
         # TODO Cache results
         py_did = DID(did) if isinstance(did, str) else did
-        for resolver in self._match_did_to_resolver(py_did):
+        did = str(py_did)
+        for resolver in await self._match_did_to_resolver(profile, py_did):
             try:
                 LOGGER.debug("Resolving DID %s with %s", did, resolver)
-                return await resolver.resolve(profile, py_did)
+                return await resolver.resolve(profile, did)
             except DIDNotFound:
                 LOGGER.debug("DID %s not found by resolver %s", did, resolver)
 
         raise DIDNotFound(f"DID {did} could not be resolved")
 
-    def _match_did_to_resolver(self, py_did: DID) -> BaseDIDResolver:
+    async def _match_did_to_resolver(
+        self, profile: Profile, py_did: DID
+    ) -> Sequence[BaseDIDResolver]:
```

Comment:
This method should just take a `did: str`.

Reply:
_match_did_to_resolver calls supports that takes a profile. how do I pass supports a profile if not through this parameter?

Reply:
done! takes a profile and a did: str.